File: app/src/main/java/com/geniusforapp/fancysample/MainActivity.java
package com.geniusforapp.fancysample;

import android.app.Dialog;
import android.os.Bundle;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import android.view.View;
import android.widget.Toast;

import com.geniusforapp.fancydialog.FancyAlertDialog;


public class MainActivity extends AppCompatActivity {

    private FancyAlertDialog.Builder dialog;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                buildAndShowDialog();
            }
        });
        setupDialogToSustainAfterRotation(fab);
    }

    private void buildAndShowDialog() {
        dialog = new FancyAlertDialog.Builder(MainActivity.this)
                .setImageDrawable(getResources().getDrawable(R.drawable.ic_cloud_computing))
                .setTextTitle("UPLOAD")
                .setTextSubTitle("128GB/500GB")
                .setBody("New DATA has ben added to the device. Do you want to sync it with the service?")
                .setPositiveButtonText("Continue")
                .setPositiveColor(R.color.colorPositive)
                .setOnPositiveClicked(new FancyAlertDialog.OnPositiveClicked() {
                    @Override
                    public void OnClick(View view, Dialog dialog) {
                        Toast.makeText(MainActivity.this, "Updating...", Toast.LENGTH_SHORT).show();
                        ((CustomApplication) getApplication()).setDialogOpened(false);
                    }
                }).setAlertFont("Roboto-Bold.ttf")
                .setBodyGravity(FancyAlertDialog.TextGravity.CENTER)
                .setTitleGravity(FancyAlertDialog.TextGravity.CENTER)
                .setSubtitleGravity(FancyAlertDialog.TextGravity.CENTER)
                .setCancelable(false)
                .build();
        ((CustomApplication) getApplication()).setDialogOpened(true);
        dialog.show();
    }

    private void setupDialogToSustainAfterRotation(View viewToPostAfter) {
        // REMEMBER to dismiss dialog in onPause method
        // Make sure that all views are drawn
        viewToPostAfter.post(new Runnable() {
            @Override
            public void run() {
                // then check if dialog was previously opened
                if (((CustomApplication) getApplication()).wasDialogOpened()) {
                    buildAndShowDialog();
                }
            }
        });
    }

    @Override
    protected void onPause() {
        super.onPause();
        if (dialog != null) dialog.dismiss();
    }
}

File: app/src/main/java/com/geniusforapp/fancysample/MainActivity.kt
package com.geniusforapp.fancysample

import android.os.Bundle
import android.view.Gravity.*
import androidx.appcompat.app.AppCompatActivity
import com.geniusforapp.fancydialog.builders.FancyDialogBuilder
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)
        buildAndShowDialog()
        fab.setOnClickListener {
            buildAndShowDialog()
        }


    }

    private fun buildAndShowDialog() {
        val dialog = FancyDialogBuilder(this, R.style.CustomDialog)
                .withCanCancel(false)
                .withImageIcon(R.drawable.endgame)
                .withTitleTypeFace(R.font.roboto_bold)
                .withSubTitleTypeFace(R.font.roboto_medium)
                .withActionPositiveTypeFace(R.font.roboto_bold)
                .withActionNegativeTypeFace(R.font.roboto_bold)
                .withTextGravity(CENTER)
                .withPanelGravity(END)
                .withTitle(R.string.dialog_text_title)
                .withSubTitle(R.string.dialog_text_message)
                .withPositive(R.string.dialog_action_yes) { view, dialog -> dialog.dismiss() }
        dialog.show()
    }

}

